### PR TITLE
PIP-205 Reorganise pom to help centralise dependency management in th…

### PR DESCRIPTION
**Pull request by jimcornmell**
Changes to pom in preparation to make BOM file for the future CVE work.  If this is approved I'll use it as a starting point to migrate Spring boot to 2.7.8.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -4,53 +4,151 @@
...
 	<parent>
 		<groupId>org.springframework.boot</groupId>
 		<artifactId>spring-boot-starter-parent</artifactId>
 		<version>2.1.3.RELEASE</version>
 	</parent>
 
+	<groupId>org.snomed.otf</groupId>
+	<artifactId>classification-service</artifactId>
+	<version>7.0.0</version>
+
```
Comment (CoderMChu) [inline pom.xml:16]: In develop it needs to be a SNAPSHOT version.

File: pom.xml [hunk 2]
```diff
@@ -4,53 +4,151 @@
...
 		<maven.compiler.target>${java.version}</maven.compiler.target>
 		<packageName>${project.artifactId}</packageName>
 		<execFinalName>${project.build.finalName}.jar</execFinalName>
+		<spring-boot.version>${project.parent.version}</spring-boot.version>
+
 		<spring-cloud-aws-version>1.2.4.RELEASE</spring-cloud-aws-version>
 		<slf4j.version>1.7.35</slf4j.version>
 		<log4j.version>2.17.2</log4j.version>
+		<snomed-owl-toolkit.version>3.0.9</snomed-owl-toolkit.version>
+		<otf-common.version>2.2.0</otf-common.version>
```
Comment (CoderMChu) [inline pom.xml:30]: It is good to have versions in one place.

File: pom.xml [hunk 3]
```diff
@@ -4,53 +4,151 @@
...
+				<groupId>org.snomed.otf</groupId>
+				<artifactId>snomed-owl-toolkit</artifactId>
+				<version>${snomed-owl-toolkit.version}</version>
+			</dependency>
+			<!-- Snomed International common -->
+			<dependency>
+				<groupId>org.ihtsdo.otf.common</groupId>
+				<artifactId>otf-common</artifactId>
+				<version>${otf-common.version}</version>
+			</dependency>
```
Comment (CoderMChu) [inline pom.xml:50]: It is a nice idea but I don't think using <dependencyManagement> tag in this service offers the same benefit as in PIP-206.

File: pom.xml [hunk 4]
```diff
@@ -4,53 +4,151 @@
...
-
-		<!-- Snomed International common -->
 		<dependency>
 			<groupId>org.ihtsdo.otf.common</groupId>
 			<artifactId>otf-common</artifactId>
-			<version>2.2.0</version>
 		</dependency>
 		<dependency>
 			<groupId>org.ihtsdo.otf.common</groupId>
 			<artifactId>resource-manager</artifactId>
```
Comment (CoderMChu) [inline pom.xml:146]: We can use the version property here which will make the pom.xml short and easy to navigate.

**Review by CoderMChu — COMMENTED** [review]
@jimcornmell I am in favour of using <dependencyManagement> tag in PIP-206 as it is a project with multiple modules. Here it doesn't offer the same benefit I think.

File: pom.xml (hunk 3 above)
Comment (jimcornmell) [reply pom.xml:50]: I agree, but this is with the idea of creating a BOM in the future, so I've made the pom's consistent.

File: pom.xml (hunk 1 above)
Comment (jimcornmell) [reply pom.xml:16]: Will do

File: pom.xml (hunk 4 above)
Comment (jimcornmell) [reply pom.xml:146]: The point is to extract the versions into a BOM

Comment (CoderMChu) [reply pom.xml:146]: Extracting them into a BOM makes sense. This will benefit more for a multiple modules project.

**Review by CoderMChu — APPROVED** [review]

**Pull request closed without merging**